tests/test_component/test_amgr.py

```diff
@@ -464,6 +481,9 @@ def test_amgr_synchronizer():
     proc.start()
     proc.join()
 
+    # Wait for AppManager to finish the message exchange
+    time.sleep(5)
```

Comment:
Why do we need this? Is this only for the test or have we introduced the need of a wait to avoid a race condition in AppManager?

Reply:
This test creates a thread/process to check *state* sync between AppManager and TaskManager. The problem is that it is too fast to terminate the thread in the test, not in the AppManager logic.

Reply:
I removed this `time.sleep` at 401a458